// shared-package/index.js
"use strict";
module.exports.deleteDynamoDb = deleteDynamoDb;
module.exports.postDynamoDb = postDynamoDb;
module.exports.sleep = sleep;

const uuid = require("uuid/v4");

function deleteDynamoDb(event, callback, dynamoDb) {
  const data = JSON.parse(event.body);
  const TableName = process.env.TABLENAME;
  const Key = { id: data.id }
  const params = {
    TableName,
    Key,
  };

  dynamoDb.delete(params, (error, result) => {
    if (error) {
      console.error(error);
      callback(null, {
        statusCode: error.statusCode || 501,
        headers: { "Content-Type": "text/plain" },
        body: "Couldn't delete the item."
      });
      return;
    }

    const response = {
      statusCode: 200,
      body: JSON.stringify(result)
    };
    callback(null, response);
  });
}

function postDynamoDb(event, callback, dynamoDb) {
  let Item = JSON.parse(event.body);
  let Item2 = { ...Item }
  let id = Item2.id
  delete Item.id;
  const TableName = process.env.TABLENAME;
  if (id===undefined) {
    id = uuid();
  }
  Item = { ...Item, id };
  const params = { TableName, Item };

  dynamoDb.put(params, error => {
    if (error) {
      console.error(error);
      callback(null, {
        statusCode: error.statusCode || 501,
        headers: { "Content-Type": "text/plain" },
        body: "Couldn't post the item."
      });
      return;
    }

    const response = {
      statusCode: 200,
      body: JSON.stringify(params.Item)
    };
    callback(null, response);
  });
}

function sleep(ms) {
  return new Promise(resolve => setTimeout(resolve, ms));
}
